actioncable/lib/action_cable/server/connection/client_socket.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+require "websocket/driver"
+
+module ActionCable
+  module Server
+    class Connection
+      #--
+      # This class is heavily based on faye-websocket-ruby
+      #
+      # Copyright (c) 2010-2015 James Coglan
+      class ClientSocket # :nodoc:
```

Comment:
I will want to use an entirely different implementation of WebSocket - is this going to be a problem?

Reply:
Nope.

The way I see Falcon integration is as follows: (almost) nothing from the `action_cable/server` folder is gonna be used. This is the code specific to Rails WebSocket server implementation. Other server are free (and encouraged) to use low-level code which works better for them.

We will probably extract some parts from the Rails server code along the way (e.g., access logs, etc.). But today, you can already start building a Falcon-specific version and use only the `action_cable/channel` and `action_cable/connection` code from Rails.